src/core/i18n/utils.js

```diff
@@ -234,22 +234,36 @@ type I18nConfig = {|
   _momentDefineLocale?: Function,
 |};
 
+type makeI18nOptions = {|
+  // FIXME this should be tied back to the Intl type.
+  _Intl?: any,
+|};
+
+
 // Create an i18n object with a translated moment object available we can
 // use for translated dates across the app.
-export function makeI18n(i18nData: I18nConfig, lang: string, _Jed: Jed = Jed) {
+export function makeI18n(
+  i18nData: I18nConfig,
+  lang: string,
+  _Jed: Jed = Jed,
+  { _Intl = Intl }: makeI18nOptions = {}
+) {
   const i18n = new _Jed(i18nData);
   i18n.lang = lang;
 
   // TODO: move all of this to an I18n class that extends Jed so that we
   // can type-check all the components that rely on the i18n object.
   // Note: the available locales for tests are controlled in tests/setup.js
+  if (typeof _Intl === 'object' && Object.prototype.hasOwnProperty.call(_Intl, 'NumberFormat')) {
+    i18n.NumberFormat = new _Intl.NumberFormat(lang);
```

Comment:
nit: the convention in our JS is to start class instances with a lower case, like `i18n.numberFormat =`. I think we have some lint rules for this too but they probably don't catch object assignments.